backend/src/main/java/ca/bc/gov/hlth/mohums/controller/ClientController.java

```diff
@@ -0,0 +1,80 @@
+package ca.bc.gov.hlth.mohums.controller;
+
+import ca.bc.gov.hlth.mohums.webclient.WebClientService;
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
+import net.minidev.json.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Mono;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+
+@RestController
+public class ClientController {
+
+    private final WebClientService webClientService;
+    private final Logger logger = LoggerFactory.getLogger(ClientController.class);
+
+    private static final JSONParser jsonParser = new JSONParser(JSONParser.DEFAULT_PERMISSIVE_MODE);
```

Comment:
In SpringBoot, services are singletons by default. It Jakarta EE we used thread-safe pools of EJBs using the `@Stateless` annotation, so we don't have to worry much about thread-safety there. SpringBoot is closer to Apache Camel in that sense, where Processors are singletons.

When dealing with singletons we always have to check the documentation to see if the class field we're adding is thread-safe. For example, a common gotcha is that Java's basic [DateFormat](https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/text/DateFormat.html) class is _not_ thread-safe: "Date formats are not synchronized. It is recommended to create separate format instances for each thread. If multiple threads access a format concurrently, it must be synchronized externally." I remember Greg Perkins finding a bug in Camel route that used DateFormat in a processor, which surfaced only during lots of concurrent requests.

So is JSONParser thread-safe? We don't know. On one hand it would be surprising if it _wasn't_ thread-safe, but on the other hand I find it very surprising that DateFormat isn't thread-safe. So [lacking documentation](http://www.javapractices.com/topic/TopicAction.do?Id=48), it's safest to just assume that the class isn't thread-safe.

The fix is easy and has no cost. Just put the JSONParser right in the method. It might affect performance, but then maybe not.

Reply:
I can't speak to the thread safety of JSONParser, so I moved it into the method in this commit -> https://github.com/bcgov/moh-keycloak-user-management/pull/27/commits/ef7900cc369a4802503feeb1805a8c79dca3f6ae